elastic/kibana
# Onboard Latency Threshold rule type with FAAD

Towards: https://github.com/elastic/kibana/issues/169867

This PR onboards Latency Threshold rule type with FAAD.

### To verify

1. Run the following script to generate APM data:
```
node scripts/synthtrace simple_trace.ts --local --live
```

2. Create a latency threshold rule.
Example:
```
POST kbn:/api/alerting/rule
{
  "params": {
    "aggregationType": "avg",
    "environment": "ENVIRONMENT_ALL",
    "threshold": 400,
    "windowSize": 5,
    "windowUnit": "m"
  },
  "consumer": "alerts",
  "schedule": {
    "interval": "1m"
  },
  "tags": [],
  "name": "testinggg",
  "rule_type_id": "apm.transaction_duration",
  "notify_when": "onActionGroupChange",
  "actions": []
}
```
3. Your rule should create an alert and should saved it in `.internal.alerts-observability.apm.alerts-default-000001`
Example:
```
GET .internal.alerts-*/_search
```
4. Set `threshold: 10000`

5. The alert should be recovered and the AAD in the above index should be updated `kibana.alert.status: recovered`.
